pkg/vault/test/foundry/utils/BaseVaultTest.sol

```diff
@@ -259,6 +259,32 @@ abstract contract BaseVaultTest is VaultStorage, BaseTest, Permit2Helpers {
         }
     }
 
+    function getBalances(address user, IERC20[] memory tokensToTrack) internal view returns (Balances memory balances) {
+        balances.userBpt = IERC20(pool).balanceOf(user);
+        balances.aliceBpt = IERC20(pool).balanceOf(alice);
+        balances.bobBpt = IERC20(pool).balanceOf(bob);
+        balances.hookBpt = IERC20(pool).balanceOf(poolHooksContract);
+        balances.lpBpt = IERC20(pool).balanceOf(lp);
+
+        balances.userTokens = new uint256[](tokensToTrack.length);
```

Comment:
It doesn't really matter because this is a test, but could define a `uint256 numTokens = tokensToTrack.length` and then use `numTokens` in all the array initializations. Maybe also document why you need this (vs. the regular `getBalances`): you're tracking tokens across multiple pools, so you can't just get the tokens for one pool.